server/server.go

```diff
@@ -652,17 +656,89 @@ func (s *Server) LoopContext() context.Context {
 
 func (s *Server) startServerLoop(ctx context.Context) {
 	s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(ctx)
-	s.serverLoopWg.Add(4)
+	s.serverLoopWg.Add(5)
 	go s.leaderLoop()
 	go s.etcdLeaderLoop()
 	go s.serverMetricsLoop()
 	go s.encryptionKeyManagerLoop()
+	go s.startGCTuner()
 	if s.IsKeyspaceGroupEnabled() {
 		s.initTSOPrimaryWatcher()
 		s.initSchedulingPrimaryWatcher()
 	}
 }
 
+func (s *Server) startGCTuner() {
+	defer logutil.LogPanic()
+	defer s.serverLoopWg.Done()
+
+	ctx, cancel := context.WithCancel(s.serverLoopCtx)
+	defer cancel()
+
+	tick := time.NewTicker(gcTunerCheckCfgInterval)
+	defer tick.Stop()
+	totalMem, err := memory.MemTotal()
```

Comment:
Does totalMem need to be updated?

Reply:
It remains the same with TiDB Tuner.  https://docs.pingcap.com/tidb/stable/system-variables/#tidb_enable_gogc_tuner-new-in-v640